src/Mafin.Wizard.Api/ResourceTemplateProvider.cs

```diff
@@ -0,0 +1,81 @@
+using System.Reflection;
+
+namespace Mafin.Wizard.Api;
+
+internal class ResourceTemplateProvider
+{
+    private readonly string[] _directories;
+    private readonly string[] _directoryWithSubdirectories;
+    private readonly Assembly _assembly;
+
+    public ResourceTemplateProvider(string[] directories, string[] directoryWithSubdirectories)
+    {
+        _assembly = typeof(TemplateReader).Assembly;
+        _directories = directories;
+        _directoryWithSubdirectories = directoryWithSubdirectories;
+
+        if (_directories is null || _directories.Length is 0)
+        {
+            _directories = [string.Empty];
+        }
+    }
+
+    private string BaseTemplatePath => _assembly.GetName().Name!;
+
+    public bool IsFullTemplateName(string value)
+    {
+        if (value == null)
+        {
+            throw new ArgumentNullException(nameof(value), "Template name cannot be null");
+        }
+
+        return value.StartsWith($"{BaseTemplatePath}.", StringComparison.OrdinalIgnoreCase)
+            && Path.HasExtension(value);
+    }
+
+    public string GetFullTemplateName(string name)
+    {
+        if (IsFullTemplateName(name))
+        {
+            return name;
+        }
+
+        var names = GetResourceNames()
+            .Where(x => Path.GetFileNameWithoutExtension(x).EndsWith(name, StringComparison.OrdinalIgnoreCase)).ToList();
+
+        if (names.Count is 0)
+        {
+            throw new FileNotFoundException($"Template {name} is not found");
+        }
+
+        if (names.Count is > 1)
+        {
+            var fileInRoot = names.Find(x =>
+                Path.GetFileNameWithoutExtension(x).Equals($"{BaseTemplatePath}.{name}", StringComparison.OrdinalIgnoreCase));
+            return fileInRoot is not null
+                ? fileInRoot
+                : throw new FileNotFoundException($"Failed to find a unique template with {name} name");
+        }
+
+        return names.Single();
```

Comment:
It looks like this code is unreachable. Could you, please, re-check?

Reply:
this exception is necessary for cases when it is impossible to unambiguously determine which template should be used.
as example you can look at `mafin-wizard-api/templates/CSharp/Mafin_Configuration_json.sbn`

```
{
  {{~
    include 'TestTypes.UI.ConfigFiles.Mafin_Configuration'
    include 'TestTypes.API.ConfigFiles.Mafin_Configuration'
    include 'Features.Configuration.ConfigFiles.Mafin_Configuration'
    configSections | array.join ','
  ~}}
} 
```

in `include` command you can pass the names of templates partially, as an example:
`include 'Mafin_Configuration'`
`include 'ConfigFiles.Mafin_Configuration'`

However, if `include 'Mafin_Configuration'` is used, three templates will be found:
`TestTypes.UI.ConfigFiles.Mafin_Configuration`
`TestTypes.API.ConfigFiles.Mafin_Configuration`
`Features.Configuration.ConfigFiles.Mafin_Configuration`
 and it will be impossible to determine which one to use